frame/nfts/src/types.rs

```diff
@@ -172,3 +176,128 @@ pub struct PriceWithDirection<Amount> {
 	/// A direction (send or receive).
 	pub(super) direction: PriceDirection,
 }
+
+/// Support for up to 64 user-enabled features on a collection.
+#[bitflags]
+#[repr(u64)]
+#[derive(Copy, Clone, RuntimeDebug, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub enum CollectionSetting {
+	/// Items in this collection are transferable.
+	TransferableItems,
+	/// The metadata of this collection can be modified.
+	UnlockedMetadata,
+	/// Attributes of this collection can be modified.
+	UnlockedAttributes,
+	/// When this isn't set then the deposit is required to hold the items of this collection.
+	DepositRequired,
+}
+pub(super) type CollectionSettings = BitFlags<CollectionSetting>;
+
+/// Wrapper type for `CollectionSettings` that implements `Codec`.
+#[derive(Clone, Copy, PartialEq, Eq, Default, RuntimeDebug)]
+pub struct CollectionConfig(pub CollectionSettings);
+
+impl CollectionConfig {
+	pub fn all_settings_enabled() -> Self {
+		Self(BitFlags::EMPTY)
```

Comment:
Normally bitflags are used in a way that a 1 signals enabled and 0 disabled.  
If you use `EMPTY` as enabled, then this will require a storage migration if you add a new setting, unless you implicitly want to force-enable it onto all deployed nfts+collections.  
1 = enabled is more defensive.

Reply:
You set 1 (true) for the disabled things, so if nothing is disabled, then there is no need to store anything.
Why would you need to run a migration for the new settings? If you add smth new, it will be enabled by default, until you explicitly disable it